File: skyspace33_ui/src/services/CityService.js
import http from "../http-common"; 

class CityService {
  getAllCitys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/city/citys`, searchDTO);
  }

  get(cityId) {
    return this.getRequest(`/city/${cityId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/city?field=${matchData}`, null);
  }

  addCity(data) {
    return http.post("/city/addCity", data);
  }

  update(data) {
  	return http.post("/city/updateCity", data);
  }
  
  uploadImage(data,cityId) {
  	return http.postForm("/city/uploadImage/"+cityId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CityService();

File: src/main/java/com/skyspace33/service/CityService.java
package com.skyspace33.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace33.domain.City;
import com.skyspace33.dto.CityDTO;
import com.skyspace33.dto.CitySearchDTO;
import com.skyspace33.dto.CityPageDTO;
import com.skyspace33.dto.CityConvertCriteriaDTO;
import com.skyspace33.service.GenericService;
import com.skyspace33.dto.common.RequestDTO;
import com.skyspace33.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CityService extends GenericService<City, Integer> {

	List<City> findAll();

	ResultDTO addCity(CityDTO cityDTO, RequestDTO requestDTO);

	ResultDTO updateCity(CityDTO cityDTO, RequestDTO requestDTO);

    Page<City> getAllCitys(Pageable pageable);

    Page<City> getAllCitys(Specification<City> spec, Pageable pageable);

	ResponseEntity<CityPageDTO> getCitys(CitySearchDTO citySearchDTO);
	
	List<CityDTO> convertCitysToCityDTOs(List<City> citys, CityConvertCriteriaDTO convertCriteria);

	CityDTO getCityDTOById(Integer cityId);







}





